plugins/glustershd/init.go

```diff
@@ -32,10 +33,26 @@ func (p *Plugin) RestRoutes() route.Routes {
 			Version:      1,
 			ResponseType: utils.GetTypeString(([]glustershdapi.BrickHealInfo)(nil)),
 			HandlerFunc:  selfhealInfoHandler},
+		route.Route{
+			Name:        "SelfHeal",
+			Method:      "POST",
+			Pattern:     "/volumes/{name}/heal",
+			Version:     1,
+			HandlerFunc: selfHealHandler},
 	}
 }
 
 // RegisterStepFuncs registers transaction step functions with
 // Glusterd Transaction framework
 func (p *Plugin) RegisterStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"selfheal.Heal", txnSelfHeal},
+	}
+
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
```

Comment:
As this is only one step function, just register directly, without a loop or a struct.